energywebfoundation/ocn-tools-elia
# Feat - ELIA90&91 - claims issuance flow

Working claims lifecycle.

**Questions**
- Should the DID and Asset classes be merged?

**Current limitations**
- Only works for vehicles
- TSO approval isn't actually approving specific request, just approving the first pending one.
- Need to specify the DID that you want to issue the claim. Perhaps it would be better to be able to specific a role that you want to be able to issue a claim (somewhat related to: https://energyweb.atlassian.net/browse/MYEN-280)

**How to run & test**
Setup:
- Use the `feat/elia90-msp-request-role` branch for the frontend (see PR: https://github.com/energywebfoundation/ev-dashboard-frontend/pull/21)
- Copy the `config.docker.ts` to `config.ts`
- Set the `prequalifcationIssuerDID` in the `config.ts` to the DID that you will be using with metamask

Run:
1. Log in as an MSP (or another other than a TSO) to the EV frontend. Select a vehicle, check "prequalified" then click "Apply to asset" button. -> You should see logs from msp-simulation saying that claim request is being created (note that the issuance NATs listener also gets triggered by the claim creation. This can be improved)
2. Log in as a TSO (with the account used for the `prequalificationIssuerDID`), select a check, check "prequalified" then click "Apply to asset" button. You'll be asked to sign twice (once to initialized iam-client-lib, once to sign the claim) -> You should see log from msp-simulation saying that an Issued Claim has been received and is being published to the asset's DID document. Optionally, you could then somehow read the DID Document and confirm that the service has been added.
